cmd/root.go

```diff
@@ -59,17 +61,62 @@ const (
 	formatDefault = "default"
 )
 
+const (
+	scorecardLong = "A program that shows security scorecard for an open source software."
+	scorecardUse  = `./scorecard --repo=<repo_url> [--checks=check1,...] [--show-details] [--policy=file]
+or ./scorecard --{npm,pypi,rubgems}=<package_name> [--checks=check1,...] [--show-details] [--policy=file]`
+	scorecardShort = "Security Scorecards"
+)
+
+func readPolicy() *spol.ScorecardPolicy {
+	if policyFile != "" {
+		data, err := os.ReadFile(policyFile)
+		if err != nil {
+			log.Fatal(err)
+		}
+		sp, err := spol.ParseFromYAML(data)
+		if err != nil {
+			log.Fatal(err)
+		}
+		return sp
+	}
+	return nil
+}
+
+func ensureChecksHavePolicies(sp *spol.ScorecardPolicy, enabledChecks checker.CheckNameToFnMap) {
+	for checkName := range enabledChecks {
+		_, exists := sp.Policies[checkName]
+		if !exists {
+			log.Fatal(fmt.Sprintf("check %s has no policy declared", checkName))
+		}
+	}
+}
+
 var rootCmd = &cobra.Command{
-	Use: `./scorecard --repo=<repo_url> [--checks=check1,...] [--show-details]
-or ./scorecard --{npm,pypi,rubgems}=<package_name> [--checks=check1,...] [--show-details]`,
-	Short: "Security Scorecards",
-	Long:  "A program that shows security scorecard for an open source software.",
+	Use:   scorecardUse,
+	Short: scorecardShort,
+	Long:  scorecardLong,
 	Run: func(cmd *cobra.Command, args []string) {
 		// UPGRADEv3: remove.
 		var v3 bool
 		if _, v3 = os.LookupEnv("SCORECARD_V3"); v3 {
 			fmt.Printf("**** Using SCORECARD_V3 code ***** \n\n")
 		}
+		if format == formatSarif && !v3 {
+			log.Fatalf("sarif not supported yet")
+		}
+
+		if policyFile != "" && !v3 {
+			log.Fatal("policy not supported yet")
+		}
+
+		var policy *spol.ScorecardPolicy
+		if policyFile != "" {
+			policy = readPolicy()
+			if policy == nil {
+				log.Fatal("empty policy")
+			}
+		}
```

Comment:
Maybe move this down to line 173 instead of repeatedly checking for `policyFile != ""`